fairseq/optim/fairseq_optimizer.py

```diff
@@ -101,11 +101,13 @@ def all_reduce_grads(self, module):
 
     def multiply_grads(self, c):
         """Multiplies grads by a constant *c*."""
+        c = float(c)
+        params_with_grad = []
         for p in self.params:
             if p.grad is not None:
-                if torch.is_tensor(c):
-                    c = c.to(p.grad.device)
-                p.grad.data.mul_(c)
+                params_with_grad.append(p.grad.data)
+        # foreach reduces gpu kernel launch
+        torch._foreach_mul_(params_with_grad, c)
```

Comment:
Are we safe with `_foreach_mul_` here? As far as I know it is not part of PyTorch's public API.